File: web/src/main/java/com/alexeiddg/web/model/Project.java
package com.alexeiddg.web.model;

import lombok.*;
import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long projectId;

    @Column(name = "project_name", nullable = false, length = 100)
    private String projectName;

    @Column(name = "project_desc", length = 500)
    private String projectDesc;

    @ManyToOne
    @JoinColumn(name = "manager_id")
    private AppUser manager;

    @ManyToMany
    @JoinTable(
            name = "developers",
            joinColumns = @JoinColumn(name = "project_id"),
            inverseJoinColumns = @JoinColumn(name = "user_id")
    )
    private Set<AppUser> developers = new HashSet<>();

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;
}

File: frontend/src/types/model/Project.ts
import {AppUser} from "@/types/model/AppUser";
import {Team} from "@/types/model/Team";
import {Sprint} from "@/types/model/Sprint";

export interface Project {
    id: number;
    projectName: string;
    projectDescription: string;
    manager: AppUser;
    teams: Team[];
    sprints: Sprint[];
    isActive: boolean;
    createdAt: string;
    updatedAt: string;
}
